src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -552,7 +554,15 @@ protected virtual bool IncrementNonce(Transaction tx, BlockHeader header, IRelea
 
                         if (unspentGas >= codeDepositGasCost)
                         {
-                            WorldState.InsertCode(env.ExecutingAccount, substate.Output, spec);
+                            var code = substate.Output.ToArray();
+                            var codeInfo = new CodeInfo(code);
+                            // Start generating the JumpDestinationBitmap in background.
+                            ThreadPool.UnsafeQueueUserWorkItem(codeInfo, preferLocal: false);
+
+                            Hash256 codeHash = code.Length == 0 ? Keccak.OfAnEmptyString : Keccak.Compute(code.AsSpan());
+                            WorldState.InsertCode(env.ExecutingAccount, codeHash, code, spec);
+                            VirtualMachine.CacheCodeInfo(codeHash, codeInfo);
```

Comment:
Should this be hidden under `WorldState.InsertCode`? If not at maybe under Extension method? If not maybe at least method here (can be local).